src/coreclr/jit/phase.cpp

```diff
@@ -232,6 +232,7 @@ void Phase::PostPhase(PhaseStatus status)
             comp->fgDebugCheckLinks();
             comp->fgDebugCheckNodesUniqueness();
             comp->fgVerifyHandlerTab();
+            comp->fgDebugCheckLoopTable();
```

Comment:
I love having this here. (In fact, I wrote this exact change back in December but never got it in :-( ).

IIRC, not too many of the optimization phases actually go down this code path, so it won't get run much before the loop table is marked as not valid. I tried expanding the number of optimization phases that were "true" phase object phases, and used post-phase processing, and they kept hitting cases where the loop table was invalid.

Reply:
VN and hoisting are dependent on proper loop structure, so it would be good to be confident that things are ok there. Let me try converting them to true phases (along with ssa and early prop)